build.gradle

```diff
@@ -53,6 +54,7 @@ subprojects {
         opencensusVersion = '0.24.0'
 
         configureProtoCompilation = {
+            boolean isAndroid = project.getName().contains('android')
```

Comment:
change seems okay. i am wondering if it is a good idea to split this into two sections; android and java.

Reply:
Configurations for android and java are mostly the same except:
1. Android needs the extra "lite" option for code generation.
2. The build unit for Java project is sourceSet while for Android is variant.

Other than those, they still shares the same configuration in common. I thought about having android's own `configureProtoCompilation` block, but it turned out that majority lines are the same as Java's.

Reply:
okay. yeah little worried about the build becoming more complicated, but splitting sounds like could be harder to maintain